shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1175,8 +1175,16 @@ - (void)keyboardWillChangeFrame:(NSNotification*)notification {
 }
 
 - (void)keyboardWillBeHidden:(NSNotification*)notification {
-  // When keyboard hide, the keyboardWillChangeFrame function will be called to update viewport
-  // metrics. So do not call [self updateViewportMetrics] here again.
+  if (self.targetViewInsetBottom != 0) {
+    // Ensure the keyboard will be dismissed. Just like the keyboardWillChangeFrame,
+    // keyboardWillBeHidden is also in an animation block in iOS sdk, so we don't need to set the
+    // animation curve. Related issue: https://github.com/flutter/flutter/issues/99951
+    self.targetViewInsetBottom = 0;
+    NSDictionary* info = [notification userInfo];
+    NSTimeInterval duration =
+        [[info objectForKey:UIKeyboardAnimationDurationUserInfoKey] doubleValue];
```

Comment:
Use subscripting.
```suggestion
    NSTimeInterval duration =
        info[UIKeyboardAnimationDurationUserInfoKey].doubleValue;
```

Reply:
Emm using subscripting will not compile successfully
``` 
error: property 'doubleValue' not found on object of type 'id _Nullable'
            info[UIKeyboardAnimationDurationUserInfoKey].doubleValue;
                                                         ^
1 error generated.
```